doc: add ScyllaDB image upgrade guides  for patch releases

annastuchlik opened the pull request:
This PR adds the missing upgrade guides for upgrading the ScyllaDB image to a patch release:

- ScyllaDB 5.0: /upgrade/upgrade-opensource/upgrade-guide-from-5.x.y-to-5.x.z/upgrade-guide-from-5.x.y-to-5.x.z-image/
- ScyllaDB Enterprise: /upgrade/upgrade-enterprise/upgrade-guide-from-2021.1-to-2022.1/upgrade-guide-from-2022.1-to-2022.1-image/ (the file name is wrong and will be fixed with another PR)

In addition, the section regarding the recommended upgrade procedure has been improved.
Fixes https://github.com/scylladb/scylladb/issues/11450
Fixes https://github.com/scylladb/scylladb/issues/11452

annastuchlik commented:
@yaronkaikov I've added you as a reviewer because the last commit includes the improvements you requested. I've made some changes to clarify that there are two alternative upgrade procedures to avoid confusion. You can review only that part. When published, it looks as follows:

![image](https://user-images.githubusercontent.com/37244380/188600851-01511176-a075-465d-b4df-007537d639b4.png)

yaronkaikov commented:
/cc @Annamikhlin

Annamikhlin commented:
I reviewed the upgrade procedure of ScyllaDB Enterprise, please find my comments:
In section "To upgrade ScyllaDB and update 3rd party and OS packages (RECOMMENDED)" - 
  1. Please remove the "New in version Scylla: 5.0" - this version is not relevant for Scylla Enterprise product.
  2. Please update "New in version Scylla**:** Enterprise 2021.1.10" to New in version Scylla Enterprise**:** 2021.1.10"
  3. In step #2 - please update the command to:
       cat scylla-**enterprise**-packages-<version>-<arch>.txt | sudo xargs -n1 apt-get -y
  4. In "<version> - The Scylla version to which you are upgrading **( 2022.1 )**." - update to **( 2022.1.y )**
  5. In the example command - please update the command to "cat scylla-**enterprise**-packages-**2022.1.10**-x86_64.txt | sudo xargs -n1 apt-get -y

annastuchlik commented:
@Annamikhlin Thanks for reviewing. The section about updating the OS packages is now a common file shared by ScyllaDB OSS and Enterprise. If there are so many differences, I need to restructure the documentation and create new includes. I'll ask you to re-review it when I'm done.

yaronkaikov commented:
@annastuchlik One more comment about the procedure. 
After the following line `Update the ScyllaDB Enterprise Repo to 2022.1` we need to add addtional step `sudo apt-get update` to load the new repo. otherwise the next command will fail to find the packages

yaronkaikov commented on docs/upgrade/_common/upgrade-image-enterprise.rst:
```diff
@@ -0,0 +1,36 @@
+Choosing this upgrade procedure allows you to upgrade your ScyllaDB version and update the 3rd party and OS packages using one command. 
+
+#. Update the |SCYLLA_REPO|_ to |NEW_VERSION|.
+
```
@annastuchlik  After the following line `Update the ScyllaDB Enterprise Repo to 2022.1` we need to add additional step `sudo apt-get update` to load the new repo. otherwise the next command will fail to find the packages

yaronkaikov commented on docs/upgrade/_common/upgrade-image-opensource.rst:
```diff
@@ -0,0 +1,36 @@
+#. Update the |SCYLLA_REPO|_ to |NEW_VERSION|.
+
+#. Run the following command to update the manifest file:
+    
```
Same comment as above

Annamikhlin commented on docs/upgrade/upgrade-enterprise/upgrade-guide-from-2022.x.y-to-2022.x.z/upgrade-guide-from-2022.x.y-to-2022.x.z-image.rst:
```diff
@@ -0,0 +1,16 @@
+.. |SCYLLA_NAME| replace:: ScyllaDB Enterprise Image
+.. |PKG_NAME| replace:: scylla
+.. |SCYLLA_REPO| replace:: ScyllaDB Enterprise deb repo
+.. _SCYLLA_REPO: https://www.scylladb.com/download/?platform=ubuntu-20.04&version=scylla-5.0
```
This repo path is relevant to ScyllaDB Open Source

Annamikhlin commented on docs/upgrade/_common/upgrade-guide-v2022-patch-ubuntu-and-debian-p2.rst:
```diff
@@ -0,0 +1,95 @@
+
+       sudo apt-get clean all
+       sudo apt-get update
+       sudo apt-get dist-upgrade scylla-enterprise
```
These step is relevant to the old/previous procedure

annastuchlik replied:
I think this step should be left as is. I was told that this is an alternative procedure - not recommended but should be included. And this is what was specified in the introduction: 

There are two alternative upgrade procedures:

- [Upgrading ScyllaDB and simultaneously updating 3rd party and OS packages](http://127.0.0.1:5500/upgrade/upgrade-enterprise/upgrade-guide-from-2022.x.y-to-2022.x.z/upgrade-guide-from-2022.x.y-to-2022.x.z-image/#upgrade-image-recommended-procedure). It is recommended if you are running a ScyllaDB official image (EC2 AMI, GCP, and Azure images), which is based on Ubuntu 20.04.
- [Upgrading ScyllaDB without updating any external packages](http://127.0.0.1:5500/upgrade/upgrade-enterprise/upgrade-guide-from-2022.x.y-to-2022.x.z/upgrade-guide-from-2022.x.y-to-2022.x.z-image/#upgrade-image-enterprise-upgrade-guide-regular-procedure).

annastuchlik commented:
In v2 I've:
- added the step to load the new repo.
- fixed the links to include the correct platform and version.
- updated the page title to use variables rather than hardcoded information.

@Annamikhlin @yaronkaikov Could you re-review it?

annastuchlik commented:
@Annamikhlin @yaronkaikov Could you re-review this PR?

yaronkaikov commented on docs/upgrade/_common/upgrade-image-enterprise.rst:
```diff
@@ -0,0 +1,42 @@
+    
+    .. code:: sh 
+    
+       cat scylla-enterprise-packages-<version>-<arch>.txt | sudo xargs -n1 apt-get -y
```
missing the install command here it should be
`cat scylla-packages-<version>-<arch>.txt | sudo xargs -n1 apt-get install -y`

Also there is an alternative solution suggested by @soyacz please refer to https://github.com/scylladb/scylladb/issues/11456

annastuchlik replied:
@yaronkaikov This (and below) are examples for ScyllaDB Enterprise, so I guess we should stick to "scylla-enterprise-packages", but add the `install` command.

yaronkaikov commented on docs/upgrade/_common/upgrade-image-enterprise.rst:
```diff
@@ -0,0 +1,42 @@
+    
+        .. code:: sh 
+           
+           cat scylla-enterprise-packages-2022.1.10-x86_64.txt | sudo xargs -n1 apt-get -y
```
Same here

yaronkaikov commented on docs/upgrade/_common/upgrade-image-opensource.rst:
```diff
@@ -0,0 +1,43 @@
+    
+    .. code:: sh 
+    
+       cat scylla-packages-<version>-<arch>.txt | sudo xargs -n1 apt-get -y
```
And here

yaronkaikov commented on docs/upgrade/_common/upgrade-image-opensource.rst:
```diff
@@ -0,0 +1,43 @@
+    
+        .. code:: sh 
+           
+           cat scylla-packages-5.1.2-x86_64.txt | sudo xargs -n1 apt-get -y
```
And here

annastuchlik commented:
@yaronkaikov I've made the update. Could you re-review it?

yaronkaikov reviewed (approved):
LGTM

nyh commented:
> @nyh Could you merge this PR? These patch upgrade guides have been missing from our documentation.

Done.

annastuchlik commented:
@nyh Thank you (for both PRs)!

annastuchlik commented:
@scylladb/scylla-maint Please backport to 5.1

denesb commented:
> @scylladb/scylla-maint Please backport to 5.1

Doesn't apply cleanly to 5.1.
